src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const findUserById = (userId: number) => {
+  return usersFromServer.find(usr => usr.id === userId);
+};
+
+const getPreparedTodos = () => {
+  return todosFromServer.map(todo => {
+    return {
+      ...todo,
+      user: findUserById(todo.userId) || null,
```

Comment:
The `findUserById` function returns `undefined` if a user is not found, which is then converted to `null`. This might be intentional, but ensure that the rest of your application can handle `null` values for `user` in todos.